### DISCUSSION TITLE
Apple OAuth2 and Flutter

### DISCUSSION THREAD

**Discussion by jonafeucht** [pocketbase/pocketbase#1968] [Q&A]
As discussed in #1954 the web OAuth2 implementation works.
I've now started to test it with Flutter and everything works fine until I try to auth with Pocketbase, where I'm getting this error:
```yaml
{
  "errorMessage": "oauth2: cannot fetch token: 400 Bad Request\nResponse: {\"error\":\"invalid_client\"}"
}
```

Here is the corresponding flutter code: https://github.com/jonafeucht/pocketbase-flutter-siwa-example/blob/main/lib/main.dart

~~And here is the corresponding [node app](https://glitch.com/edit/#!/quiet-foregoing-cathedral?path=server.js), which is just the one provided by the [sign_in_with_apple](https://pub.dev/packages/sign_in_with_apple) package.~~

Maybe related to: https://github.com/pocketbase/pocketbase/discussions/1890

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what you are doing. Why do you need a node-app in the first place?

The general flow should be similar to the web example - https://pocketbase.io/docs/authentication/#web-oauth2-integration, aka. you need 2 handlers:

1. Somewhere to show the "Login with ..." link(s)

2. Somewhere to handle the provider redirect (eg. with a deeplink) and send the `authWithOAuth2()` call.

In the above flutter app it seems that you are calling the `authWithOAuth2()` before even calling the `signInWithAppleEndpoint`.

But again, I'm not sure why do you need the node action in the first place. The redirect should be handled directly by the client (that's why `response_mode` must be `query`).

_The OAuth2 flow in general will be simplified in #55 and deeplinks will not be necessary._

**Reply by ganigeorgiev** [reply]
One more thing that I probably should document, additional scopes for the Apple provider are not supported because the redirect will be always POST in this case (at least based on their documentation). This will be resolved with #55.

**Reply by jonafeucht** [reply]
The code uses the native sign in flow. I switch to https://pub.dev/packages/flutter_web_auth_2, which now opens up a web browser and now it works.
